test_helpers/pool_helper.go

```diff
@@ -215,16 +256,23 @@ func SetClusterRO(dialers []tarantool.Dialer, connOpts tarantool.Opts,
 		return fmt.Errorf("number of servers should be equal to number of roles")
 	}
 
+	ctx, cancel := GetConnectContext()
+	defer cancel()
+
+	// Apply roles in parallel.
+	errs := make([]error, len(dialers))
+	var wg sync.WaitGroup
+	wg.Add(len(dialers))
 	for i, dialer := range dialers {
-		ctx, cancel := GetConnectContext()
-		err := SetInstanceRO(ctx, dialer, connOpts, roles[i])
-		cancel()
-		if err != nil {
-			return err
-		}
+		// Pass loop variables to avoid its closure.
+		go func(i int, dialer tarantool.Dialer) {
+			defer wg.Done()
+			errs[i] = SetInstanceRO(ctx, dialer, connOpts, roles[i])
+		}(i, dialer)
```

Comment:
```suggestion
		go func() {
			defer wg.Done()
			errs[i] = SetInstanceRO(ctx, dialer, connOpts, roles[i])
		}()
```
It's just a cosmetic, so up to you.
Since Go 1.22 it's safe to capture variables when using closures with concurrency inside loop body.

Reply:
I know, but in go.mod there is line 

> go 1.20